### PR TITLE
ROC 5194 notify email for defendant on claimant response rejection of states paid or part admission

### PR DESCRIPTION
**Pull request by tharacka**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-5194

### Change description ###
Add a notification email for when claimant rejects a states paid or part admission

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/config/properties/notifications/EmailTemplates.java [hunk 1]
```diff
@@ -111,4 +111,7 @@
 
     @NotEmpty
     private String defendantAdmissionResponseToClaimant;
+
+    @NotEmpty
+    private String claimantRejectedPartAdmitOrStatesPaidEmailToDefendant;
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/config/properties/notifications/EmailTemplates.java]: As there is no different email template for these, do we really need to say what type of defendant's response claimant is rejecting in name of this field? May be just  `claimantRejectedEmailToDefendant`

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/notifications/NotificationToDefendantService.java [hunk 2]
```diff
@@ -48,6 +51,25 @@ public void notifyDefendant(Claim claim) {
         );
     }
 
+    public void notifyDefendantOfRejection(Claim claim) {
+        Response response = claim.getResponse().orElseThrow(IllegalArgumentException::new);
+
+        if (isResponseStatesPaid(response) || response.getResponseType() == ResponseType.PART_ADMISSION) {
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/notifications/NotificationToDefendantService.java]: You are adding two level of filtering to choose ` notifyDefendant(claim);`  once in EventListener `sendNotificationToDefendant(ClaimantResponseEvent event)` and then here, Please move this condition in event listener if at all needed. 

This condition might be un-necessary as well, please verify.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/config/properties/notifications/EmailTemplates.java (hunk 1 above)
Comment (tharacka) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/config/properties/notifications/EmailTemplates.java]: As the wording on the GOV Notify template is very specific I have done this. The fear is that if they change the template to be more specific to States Paid/Part Admit as opposed to the general format it is at the moment. I agree with 'claimantRejectedEmailToDefendant' if the template wasn't specific.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/notifications/NotificationToDefendantService.java (hunk 2 above)
Comment (singh-surjeet) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/notifications/NotificationToDefendantService.java]: I agree
AC states rejection of part admission or states paid. Please note even full defence states paid less than claim amount is sent to claim-store as part-admission states paid hence all of the logic can be put at one place.

**Pull request merged**